util/run.bzl

```diff
@@ -0,0 +1,263 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Rules to run a command inside a container, and either commit the result
+to new container image, or extract specified targets to a directory on
+the host machine.
+"""
+
+load(
+    "@io_bazel_rules_docker//container:bundle.bzl",
+    "container_bundle",
+)
+
+def _extract_impl(ctx):
+    # Since we're always bundling/renaming the image in the macro, this is valid.
+    load_statement = 'docker load -i %s' % ctx.file.image_tar.path
+
+    script = ctx.new_file(ctx.label.name + ".build")
+
+    # Generate a shell script to execute the run statement
+    ctx.actions.expand_template(
+        template=ctx.file._extract_tpl,
+        output=script,
+        substitutions={
+          "%{load_statement}": load_statement,
+          "%{flags}": " ".join(ctx.attr.flags),
```

Comment:
nit: drop flags until someone asks for it?

Reply:
I thought we had decided we'd leave it in, but I can remove it if you think we should. I personally don't really think it's necessary.